Sources/SourceControl/RepositoryManager.swift

```diff
@@ -30,8 +31,8 @@ public protocol RepositoryManagerDelegate: class {
 }
 
 public extension RepositoryManagerDelegate {
-    func fetchingWillBegin(handle: RepositoryManager.RepositoryHandle) {}
-    func fetchingDidFinish(handle: RepositoryManager.RepositoryHandle, error: Swift.Error?) {}
+    func fetchingWillBegin(handle: RepositoryManager.RepositoryHandle, fetchDetails: RepositoryManager.FetchDetails?) {}
```

Comment:
Instead of removing `func fetchingWillBegin(handle: RepositoryManager.RepositoryHandle) {}` altogether, I'd mark it deprecated and implement `func fetchingWillBegin(handle:fetchDetails:)` to call it in the extension.  This lets clients that haven't yet implemented it keep working.

This will all be made easier once there is a semantically versioned API for libSwiftPM.  Right now it's a hassle but changing the API without staging causes a lot of downstream problems for any clients.